src/components/CheckoutModal/index.js

```diff
@@ -0,0 +1,285 @@
+import styled from "styled-components";
+import { useState} from "react";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import ElectricBoltOutlinedIcon from "@mui/icons-material/ElectricBoltOutlined";
+import AttachMoneyIcon from "@mui/icons-material/AttachMoney";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableRow from "@mui/material/TableRow";
+import Paper from "@mui/material/Paper";
+import { Checkbox, CircularProgress } from "@mui/material";
+import Image from "../Main/Image";
+
+import { SignupSchema } from "../../utils/validations/validationSchema";
+import { validationRules } from "./../../utils/validations/validation";
+import axios from "axios";
+
+const CheckoutModal = ({
+    isOpen,
+    handleClose,
+    orderSummary,
+    orderedPrice,
+    handleFastDelivery,
+    clearBurger
+}) => {
+
+    const [checked, setChecked] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const [orderStatus, setOrderStatus] = useState(false); 
+
+    const handleChange = (event) => {
+        handleFastDelivery(event.target.checked);
+        setChecked(event.target.checked);
+    };
+
+    const burgerIngredients = Object.entries(orderSummary).filter((product) => product[1] > 0);
+    const createData = (name, amount) => {
+        return {name, amount};
+    };
+
+    const rows = burgerIngredients.map((ingredient) =>
+    createData(ingredient[0], ingredient[1]));
+
+    const handleOrder = async ( data )  => {
+        const {
+            name: orderName,
+            phone: orderPhone,
+            email: orderEmail,
+            address: orderAddress,
+        } = data;
+
+        const orderData = {
+            orderName,
+            orderPhone,
+            orderEmail,
+            orderAddress,
+            orderedPrice,
+            orderProduct: Object.fromEntries(burgerIngredients),
+            orderFastDelive: checked,
+        };
+
+        try {
+            setLoading(true);
+            await axios.post("https://burger-api-xcwp.onrender.com/orders",orderData);
```

Comment:
It does not work

https://user-images.githubusercontent.com/40695369/211357944-e28ff83b-ae4b-4461-b1c6-49040b129da7.mov